projects/map-ol/src/lib/map-ol.component.spec.ts

```diff
@@ -62,4 +64,51 @@ describe('MapOlComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should update a vector-layer', (done) => {
+    const vectorLayer = new VectorLayer({
+      id: 'vectorLayer',
+      name: 'vectorLayer',
+      type: 'geojson',
+      data: {
+        type: 'FeatureCollection',
+        features: [{
+            type: 'Feature',
+            properties: {},
+            geometry: {
+              type: 'Point',
+              coordinates: [11.71142578125, 46.5739667965278]
+            }
+          }]
+      }
+    });
+    component.layersSvc.addLayer(vectorLayer, 'Layers');
+
+    // layer has been added successfully
+    expect(component.layersSvc.getLayerById(vectorLayer.id)).toBeTruthy();
+    // layer has but one feature
+    expect((component.layersSvc.getLayerById(vectorLayer.id) as VectorLayer).data.features.length).toEqual(1);
+
+    setTimeout(() => { // After observables have been updated ...
+      // layer is present on map
+      const layer = component.map.getLayers().getArray()[1].getLayersArray()[0] as olVectorLayer;
+      expect(layer).toBeTruthy();
+      expect(layer.getProperties()['id']).toEqual(vectorLayer.id);
+      expect(layer.getSource().getFeatures().length).toEqual(1);
+
+      // updating data
+      vectorLayer.data = testFeatureData;
+      // the layers new data has *not yet* been passed through to the ol-layer.
+      // this ensures that there is no spooky-action-at-a-distance.
+      expect(layer.getSource().getFeatures().length).toEqual(1);
+
+      setTimeout(() => { // After observables have been updated ...
+        component.layersSvc.updateLayer(vectorLayer);
+        // now, after calling `updateLayer`, the data is present on the ol-layer
+        expect((component.map.getLayers().getArray()[1].getLayersArray()[0] as olVectorLayer).getSource().getFeatures().length).toEqual(4);
+        done();
+      }, 0);
+    }, 0);
```

Comment:
Is this working with time 0?

Reply:
Another good catch. The timeout was intended to wait for another JS cycle, because I had falsely assumed that the reaction to `addLayer` would happen asynchronously. However, there is indeed no need to wait for a cycle - so the timeouts have been removed.